src/components/ComboButton/ComboButton.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file Combo button.
+ * @copyright IBM Security 2019
+ */
+
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useRef, useState } from 'react';
+import ChevronDown16 from '@carbon/icons-react/lib/chevron--down/16';
+import ChevronUp16 from '@carbon/icons-react/lib/chevron--up/16';
+import { settings } from 'carbon-components';
+
+import Button from '../Button';
+import OverflowMenu from '../OverflowMenu';
+import OverflowMenuItem from '../OverflowMenuItem';
+import { TooltipDirection } from '../IconButton/IconButton';
+
+import { namespace as buttonNamespace } from '../Button/Button';
+
+import { getComponentNamespace } from '../../globals/namespace';
+
+export const namespace = getComponentNamespace('combo-button');
+
+const { prefix } = settings;
+
+const ComboButton = ({ children, className, direction }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const wrapper = useRef(null);
+
+  const getMenuOffset = () => {
+    const { top } = wrapper.current.getBoundingClientRect();
+    const isTop = direction === TooltipDirection.TOP;
+    return {
+      top: isTop ? top : top * -1,
+      left: 'auto',
+    };
+  };
+
+  const childrenArray = React.Children.toArray(children);
+
+  // Save first child (e.g., primary action) to use as a `Button`:
+  const primaryAction = React.cloneElement(childrenArray[0]);
```

Comment:
I wonder if you need to actually clone the element first or if you can just use `childrenArray[0]`?
```suggestion  
  const primaryActionWithProps = [childrenArray[0]].map(
```